streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -250,56 +277,95 @@ private static void addRecord(final ConsumerRecord<byte[], byte[]> record,
         }
     }
 
-    private static boolean receivedAllRecords(final Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> supersetExpectedRecords,
-                                              final Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> receivedRecords,
-                                              final Map<TopicPartition, Long> committedOffsets) {
-        if (supersetExpectedRecords == null
-            || receivedRecords == null
-            || supersetExpectedRecords.keySet().size() < committedOffsets.keySet().size()
-            || receivedRecords.keySet().size() < committedOffsets.keySet().size()) {
+    private static boolean verifyTopic(final String topic,
+                                       final boolean withRepartitioning) {
+        final boolean validTopic = "data".equals(topic) || "echo".equals(topic) || "min".equals(topic) || "sum".equals(topic);
+
+        if (withRepartitioning) {
+            return validTopic || "repartition".equals(topic) || "max".equals(topic) || "cnt".equals(topic);
+        }
+
+        return validTopic;
+    }
 
+    private static boolean receivedAllRecords(final Map<String, Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>>> receivedRecords,
+                                              final Map<TopicPartition, Long> committedOffsets,
+                                              final boolean withRepartitioning) {
+        if (receivedRecords == null) {
             return false;
         }
 
-        for (final Map.Entry<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> partitionRecords : receivedRecords.entrySet()) {
-            final TopicPartition tp = partitionRecords.getKey();
-            final int numberOfReceivedRecords = partitionRecords.getValue().size();
-            final Long committed = committedOffsets.get(new TopicPartition("data", tp.partition()));
-            if (committed != null) {
-                if (numberOfReceivedRecords < committed) {
+        // data -> data, echo, min, sum (repartition)
+        // repartition -> max, cnt
+
+        final Map<String, List<String>> inputToOutputTopics = new HashMap<>();
+        {
```

Comment:
I used it to scope the initialization for `inputToOutputTopics` -- this make the code more readable IMHO as you can easily see where the initialization code "ends".